app/lib/katello/concerns/base_template_scope_extensions.rb

```diff
@@ -206,18 +206,15 @@ def load_errata_applications(filter_errata_type: nil, include_last_reboot: 'yes'
         search_up_to = up_to.present? ? "ended_at < \"#{up_to}\"" : nil
         search_since = since.present? ? "ended_at > \"#{since}\"" : nil
         search_result = status.present? && status != 'all' ? "result = #{status}" : nil
-        search = [search_up_to, search_since, search_result].compact.join(' and ')
-
+        labels = 'label ^ ("Actions::Katello::Host::Erratum::Install", "Actions::Katello::Host::Erratum::ApplicableErrataInstall")'
         if Katello.with_remote_execution?
-          condition = ["state = 'stopped' AND ((label = 'Actions::RemoteExecution::RunHostJob' AND templates.id = ?) " \
-                       "OR label = 'Actions::Katello::Host::Erratum::Install' OR label = 'Actions::Katello::Host::Erratum::ApplicableErrataInstall')",
-                       RemoteExecutionFeature.feature('katello_errata_install').job_template_id]
-        else
-          condition = "state = 'stopped' AND (label = 'Actions::Katello::Host::Erratum::Install' OR label = 'Actions::Katello::Host::Erratum::ApplicableErrataInstall')"
+          new_labels = 'label = "Actions::RemoteExecution::RunHostJob" AND remote_execution_feature.label = katello_errata_install'
+          labels = [labels, new_labels].map { |label| "(#{label})" }.join(' OR ')
         end
 
+        search = [search_up_to, search_since, search_result, "state = 'stopped'", label_search].compact.join(' and ')
+
         tasks = load_resource(klass: ForemanTasks::Task,
-                              where: condition,
                               permission: 'view_foreman_tasks',
                               joins: 'LEFT OUTER JOIN template_invocations ON foreman_tasks_tasks.id = template_invocations.run_host_job_task_id LEFT OUTER JOIN templates ON template_invocations.template_id = templates.id',
```

Comment:
How does the join work when REX is not available? I'd expect the database would complain about `template_invocations` table not being there.